src/consumer.ts

```diff
@@ -270,9 +271,13 @@ export class Consumer extends EventEmitter {
                 err.message = `Unexpected message handler failure: ${err.message}`
             }
 
+            const messages = getMessagesByGroupId(this.pendingMessages, message)
+
             // processing has failed, remove all following messages with the same groupId
             this.pendingMessages = filterOutByGroupId(this.pendingMessages, message)
 
+            await this.changeVisibilityTimeoutOfBatch(messages, this.visibilityTimeout, 0)
```

Comment:
I would move this before the modifying of pendingMessages. Idk, how exactly are errors handled here but if it fails with error I think it would be better to keep the pendingMessages unmodified

Reply:
The errors are basically skipped in the process. There is `catch` and it emits `error` event.

Reply:
No o co mi jde je, ze tohle je async code v catch bloku, kterej muze teoreticky taky failnout (aws downtime nebo neco) a bylo by podle me lepsi si to pendingMessages nemodifikovat, pokud si nejses jistej, ze se povedlo ten visibilityTimeout zmenit 🤔

Reply:
no .. to je docela dobrá otázka .. tam se celkově všechny chyby řeší tak, že se to odchytí a jenom to emituje `error` a normálně to nechá ten kód procesovat dál .. to by možná chtělo projít celkově ten processing chyb 😐